### PR TITLE
journald: make level mappings configurable

### PR DESCRIPTION
**Pull request by kaffarell**
This allows to manually map tracing levels to journald levels.

It seems that @little-dude, who started the original PR, doesn't have time to finish this, so I picked it up.  Reapplied the changes to the newest master branch and fixed the latest comments/issues.

This will also fix/close:
- #2649
- #2661
- #2347 (the original pr)

### CODE REVIEW

File: tracing-journald/src/lib.rs [hunk 1]
```diff
@@ -131,6 +135,41 @@ impl Subscriber {
...
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use tracing_journald::{Priority, PriorityMappings};
+    /// use tracing_subscriber::prelude::*;
+    /// use tracing::error;
+    ///
+    ///let registry = tracing_subscriber::registry();
+    ///match tracing_journald::subscriber() {
```
Comment (Ralith) [inline tracing-journald/src/lib.rs:148]: Inconsistent indentation here.

**Review by Ralith — COMMENTED** [review]
LGTM aside from a cosmetic nit.

File: tracing-journald/src/lib.rs (hunk 1 above)
Comment (kaffarell) [reply tracing-journald/src/lib.rs]: Fixed it. Thanks for the review!

**Review by Ralith — APPROVED** [review]

**Comment by hongquan** [comment]
Please process this.

**Review by hawkw — APPROVED** [review]
looks good to me, thank you!

**Pull request merged**